contracts/authorities/BondingCurveAuthority.sol

```diff
@@ -0,0 +1,397 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity 0.8.20;
+
+import { Party } from "../party/Party.sol";
+import { PartyGovernanceNFT, PartyGovernance } from "../party/Party.sol";
+import { PartyFactory } from "../party/PartyFactory.sol";
+import { IERC721 } from "../tokens/IERC721.sol";
+import { MetadataProvider } from "../renderers/MetadataProvider.sol";
+import { LibSafeCast } from "contracts/utils/LibSafeCast.sol";
+
+contract BondingCurveAuthority {
+    using LibSafeCast for uint256;
+
+    error InvalidMessageValue();
+    error Unauthorized();
+    error InvalidCreatorFee();
+    error InvalidTreasuryFee();
+    error InvalidPartyDaoFee();
+    error PartyNotSupported();
+    error InvalidTotalVotingPower();
+    error ExecutionDelayTooShort();
+
+    event TreasuryFeeUpdated(uint16 previousTreasuryFee, uint16 newTreasuryFee);
+    event PartyDaoFeeUpdated(uint16 previousPartyDaoFee, uint16 newPartyDaoFee);
+    event CreatorFeeUpdated(uint16 previousCreatorFee, uint16 newCreatorFee);
+    event PartyDaoFeesClaimed(uint96 amount);
+    event PartyCardsBought(
+        Party indexed party,
+        address indexed buyer,
+        uint256[] tokenIds,
+        uint256 totalPrice,
+        uint256 partyDaoFee,
+        uint256 treasuryFee,
+        uint256 creatorFee
+    );
+    event PartyCardsSold(
+        Party indexed party,
+        address indexed seller,
+        uint256[] tokenIds,
+        uint256 sellerProceeds,
+        uint256 partyDaoFee,
+        uint256 treasuryFee,
+        uint256 creatorFee
+    );
+
+    /// @notice Info for each party controlled by this authority
+    mapping(Party => PartyInfo) public partyInfos;
+    /// @notice The global party dao fee basis points
+    uint16 public partyDaoFeeBps;
+    /// @notice The global treasury fee basis points
+    uint16 public treasuryFeeBps;
+    /// @notice The global creator fee basis points
+    uint16 public creatorFeeBps;
+    /// @notice The amount of party dao fees claimable
+    uint96 public partyDaoFeeClaimable;
+
+    /// @notice The intrinsic voting power of party cards minted by this contract
+    uint96 private constant PARTY_CARD_VOTING_POWER = uint96(0.1 ether);
+    address payable private immutable PARTY_DAO;
+    uint16 private constant BPS = 10_000;
+    uint16 private constant MAX_CREATOR_FEE = 250;
+    uint16 private constant MAX_TREASURY_FEE = 1000;
+    uint16 private constant MAX_PARTY_DAO_FEE = 250;
+    uint40 private constant MIN_EXECUTION_DELAY = 3 days;
+
+    struct BondingCurvePartyOptions {
```

Comment:
Should add comments to `BondingCurvePartyOptions` and `PartyInfo` explaining struct fields